flutterproject/version_constraint.go

```diff
@@ -0,0 +1,59 @@
+package flutterproject
+
+import (
+	"fmt"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+type VersionConstraintSource string
+
+const (
+	PubspecLockVersionSource VersionConstraintSource = "pubspec_lock"
```

Comment:
Source shouldn't be used as tracking keys, define the source - tracking key mapping in the tracker package